src/agentic/run_manager.py

```diff
@@ -119,10 +119,8 @@ def init_run_tracking(
         user_id=user_id,
         db_path=db_path
     )
-    agent._agent.set_callback.remote('handle_event', run_manager.handle_event)
```

Comment:
Using `_agent` like this isn't safe because the proxy will manage multiple agents.